src/tiled/mainwindow.cpp

```diff
@@ -498,10 +499,15 @@ void MainWindow::commitData(QSessionManager &manager)
 
 void MainWindow::closeEvent(QCloseEvent *event)
 {
-    writeSettings();
-
     if (confirmAllSave())
+    {
```

Comment:
Didn't really understand the last part, should it be like this or...
```
if (statement) {
command;
anothercommand;
}
```
...this?
```
if (statement) {
command;
anothercommand; }
```